### PR TITLE
Add group support for convolutions

### PR DESCRIPTION
**Pull request by pxl-th**
This PR adds support for groups in convolutions. It relies on [this](https://github.com/FluxML/NNlib.jl/pull/289) PR, so I'll mark it as draft for now.

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
Does this need to be rebased on NNlib?

**Comment by pxl-th** [comment]
Not sure why?

**Comment by DhairyaLGandhi** [comment]
Sorry I misspoke, I didn't mean rebase, but depend on nnlib#master for ci to pick up the latest code. Otherwise LGTM!

**Comment by pxl-th** [comment]
Also, should there be a release for NNlib, before this PR can be merged?

**Review by DhairyaLGandhi — APPROVED** [review]

File: test/conv.jl [hunk 1]
```diff
@@ -36,19 +38,19 @@ using NNlib: DenseConvDims
...
-        function NNlibCUDA.∇conv_bias!(db, dy)
-            db .= sum(dy, dims=1:(ndims(dy)-2))
-            return db
-        end
-        gputest(NNlibCUDA.∇conv_bias!, db, dy, checkgrad=false)
+    # CPU implementation of ∇conv_bias!
+    db = zeros(Float64, 1, 1, 3, 1)
+    dy = randn(Float64, 8, 8, 3, 1)
+    function NNlibCUDA.∇conv_bias!(db, dy)
+        db .= sum(dy, dims=1:(ndims(dy)-2))
```
Comment (DhairyaLGandhi) [inline test/conv.jl:52]:
```suggestion
        db .= sum(dy, dims = 1:(ndims(dy)-2))
```

Am I missing something or is this function not testing anything related to nnlib or conv or bias. Perhaps testing that the GPU kernel exists?

Comment (pxl-th) [reply test/conv.jl:52]: Seems so. I just moved it out of the loop, since it does not depend on it.

**Comment by CarloLucibello** [comment]
NNlib 0.7.25 is in the process of being registered

File: test/conv.jl [hunk 2]
```diff
@@ -36,19 +38,19 @@ using NNlib: DenseConvDims
...
-        end
-        gputest(NNlibCUDA.∇conv_bias!, db, dy, checkgrad=false)
+    # CPU implementation of ∇conv_bias!
+    db = zeros(Float64, 1, 1, 3, 1)
+    dy = randn(Float64, 8, 8, 3, 1)
+    function NNlibCUDA.∇conv_bias!(db, dy)
+        db .= sum(dy, dims = 1:(ndims(dy)-2))
+        return db
     end
+    gputest(NNlibCUDA.∇conv_bias!, db, dy, checkgrad=false)
```
Comment (DhairyaLGandhi) [inline test/conv.jl:55]: would you mind removing the function and test from here?

Comment (CarloLucibello) [reply test/conv.jl:55]: the function is needed to test the gpu method of the same function we have in NNlibCUDA

File: test/conv.jl [hunk 3]
```diff
@@ -36,19 +38,19 @@ using NNlib: DenseConvDims
...
 
-        # CPU implementation of ∇conv_bias!
-        db = zeros(Float64, 1, 1, 3, 1)
-        dy = randn(Float64, 8, 8, 3, 1)
-        function NNlibCUDA.∇conv_bias!(db, dy)
-            db .= sum(dy, dims=1:(ndims(dy)-2))
-            return db
-        end
-        gputest(NNlibCUDA.∇conv_bias!, db, dy, checkgrad=false)
+    # CPU implementation of ∇conv_bias!
```
Comment (CarloLucibello) [inline test/conv.jl:48]:
```suggestion
    ## CPU implementation of ∇conv_bias!
```

**Comment by CarloLucibello** [comment]
I had to add a dummy commit to retrigger buildkite

**Comment by CarloLucibello** [comment]
buildkite didn't pick up the new NNlib version

**Comment by pxl-th** [comment]
I've update the Project.toml, but I cannot install NNlib 0.7.25. It says the lates available is 0.7.24. Tried to update & redownload the registry but that didn't help

File: test/conv.jl (hunk 2 above)
Comment (pxl-th) [reply test/conv.jl]: So do we keep the test and function?)

**Pull request merged**